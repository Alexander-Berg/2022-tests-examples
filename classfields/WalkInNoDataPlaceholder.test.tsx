import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MockDate from 'mockdate';

import contextMock from 'autoru-frontend/mocks/contextMock';
import routerMock from 'autoru-frontend/mocks/routerMock';

import UniversalLink from 'auto-core/react/components/common/UniversalLink/UniversalLink';

import WalkInNoDataPlaceholder from './WalkInNoDataPlaceholder';

afterEach(() => {
    MockDate.reset();
});

const context = { ...contextMock, router: routerMock };

it('должен формировать правильные ссылки с мерджем параметров', () => {
    MockDate.set('2015-01-12');

    const tree = shallow(
        <WalkInNoDataPlaceholder
            routeParams={{ from: '2015-01-12', to: '2015-01-12', page: 1 }}
        />,
        { context },
    );

    const links = tree.find(UniversalLink);

    expect(shallowToJson(links)).toMatchSnapshot();
});
