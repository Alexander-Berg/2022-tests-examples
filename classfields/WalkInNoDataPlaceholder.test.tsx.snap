// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен формировать правильные ссылки с мерджем параметров 1`] = `
Array [
  <UniversalLink
    isSPA={true}
    link="linkCabinet/walk-in/?from=2015-01-11&to=2015-01-11&page=1"
  >
    день
  </UniversalLink>,
  <UniversalLink
    isSPA={true}
    link="linkCabinet/walk-in/?from=2015-01-05&to=2015-01-12&page=1"
  >
    неделю
  </UniversalLink>,
  <UniversalLink
    isSPA={true}
    link="linkCabinet/walk-in/?from=2014-12-12&to=2015-01-12&page=1"
  >
    месяц
  </UniversalLink>,
]
`;
